src/backend/db/database_manager.py

```diff
@@ -418,5 +420,421 @@ def delete_custom_field(self, customer_guid, field_name):
+                "status": ticket_result[5],
+                "custom_fields": {}
+            }
+
+            # Retrieve custom field values for the ticket
+            custom_field_query = '''
+                SELECT * 
+                FROM custom_field_values
+                WHERE ticket_id = :ticket_id
+            '''
+            custom_field_result = session.execute(
+                text(custom_field_query),
+                {"ticket_id": ticket_id},
+            ).fetchone()
+
+            if custom_field_result:
+                columns = custom_field_result.keys()
+                for column_name, value in zip(columns, custom_field_result):
+                    if column_name != "ticket_id" and value is not None:
+                        ticket_data["custom_fields"][column_name] = value
+
+            return ticket_data
+
+        except (OperationalError, DatabaseError) as e:
+            logger.error(f"Database error: {e}")
+            raise Exception("Database connectivity issue")
+        except SQLAlchemyError as e:
+            logger.error(f"SQLAlchemy error: {e}")
+            raise Exception("Database query error")
+        finally:
+            session.close()
+
+    def get_tickets_by_chat_id(self, customer_guid, chat_id):
+        customer_db_name = self.get_customer_db(customer_guid)
+        try:
+            session = DatabaseManager._session_factory()
+            # Add retry logic here for connection
+        except OperationalError as e:
+            logger.error(f"Database connection failed: {e}")
+            return {"status": "db_unreachable", "reason": "Database is currently unreachable"}
+        try:
+                # Check if the database exists
+            logger.debug(f"Checking if database {customer_db_name} exists")
+            result = session.execute(
+                text("SELECT SCHEMA_NAME FROM INFORMATION_SCHEMA.SCHEMATA WHERE SCHEMA_NAME = :db_name"),
+                {"db_name": customer_db_name}
+            ).fetchone()
+            if not result:
+                raise ValueError(f"Database {customer_db_name} does not exist")
+
+            logger.debug(f"Switching to customer database: {customer_db_name}")
+            session.execute(text(f"USE `{customer_db_name}`"))
+
+            # Check if the chat_id exists in the chat_messages table
+            chat_exists = session.execute(
+                    text("SELECT 1 FROM chat_messages WHERE chat_id = :chat_id LIMIT 1"),
+                    {"chat_id": chat_id}
+            ).fetchone()
+
+            if not chat_exists:
+                raise ValueError(f"Invalid chat_id: {chat_id} does not exist.")
+
+            query = '''SELECT ticket_id, chat_id, title, description, priority, status
+                FROM tickets WHERE chat_id = :chat_id'''
+            results = session.execute(
+                text(query),
+                {"chat_id": chat_id},
+            ).fetchall()
+            return [
```

Comment:
Can we use result.keys() to construct the json instead of using result[0], result[2] or result[5]

Reply:
modified sir